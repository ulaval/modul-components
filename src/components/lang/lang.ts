import Vue from 'vue';
import { PluginObject } from 'vue';
import Component from 'vue-class-component';
import { Prop } from 'vue-property-decorator';
import WithRender from './lang.html';
import { I18N_NAME } from '../component-names';
import { ModulVue } from '../../utils/vue';

@WithRender
@Component
export class MLang extends ModulVue {
    public text: string = '';

    @Prop()
    public msg: string;

    public mounted() {
        this.text = this.$i18n.translate(this.msg);
    }
}

const LangPlugin: PluginObject<any> = {
    install(v, options) {
        v.component(I18N_NAME, MLang);
    }
};

export default LangPlugin;
