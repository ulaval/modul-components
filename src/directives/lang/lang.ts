import Vue, { VNodeDirective } from 'vue';
import { PluginObject } from 'vue';
import { I18N_NAME } from '../directives-names';
import Messages from '@/utils/i18n';

export class MI18n extends Vue {
    public bind(element: HTMLElement, binding: VNodeDirective) {
        element.innerText = Messages.translate(binding.value);
    }
}

const MI18nPlugin: PluginObject<any> = {
    install(v, options) {
        v.directive(I18N_NAME, new MI18n());
    }
};

export default MI18nPlugin;
