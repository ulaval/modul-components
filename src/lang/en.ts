import { PluginObject } from 'vue';
import { BundleMessagesMap, ENGLISH, Messages } from '../utils/i18n/i18n';

const EnglishPlugin: PluginObject<any> = {
    install(v, options): void {
        v.prototype.$log.debug('EnglishPlugin', 'plugin.install');
        const i18n: Messages = (v.prototype).$i18n;
        if (i18n) {
            const msgs: BundleMessagesMap[] = [
                require('../components/accordion/accordion.lang.en.json'),
                require('../components/accordion-group/accordion-group.lang.en.json'),
                require('../components/list-item/list-item.lang.en.json'),
                require('../components/modal/modal.lang.en.json'),
                require('../components/chip/chip.lang.en.json'),
                require('../components/datefields/datefields.lang.en.json'),
                require('../components/datepicker/datepicker.lang.en.json'),
                require('../components/daterangepicker/daterangepicker.lang.en.json'),
                require('../components/dropdown/dropdown.lang.en.json'),
                require('../components/dropdown-group/dropdown-group.lang.en.json'),
                require('../components/overlay/overlay.lang.en.json'),
                require('../components/file-select/file-select.lang.en.json'),
                require('../components/file-upload/file-upload.lang.en.json'),
                require('../components/icon-file/icon-file.lang.en.json'),
                require('../components/inplace-edit/inplace-edit.lang.en.json'),
                require('../components/input-style/input-style.lang.en.json'),
                require('../components/link/link.lang.en.json'),
                require('../components/limit-text/limit-text.lang.en.json'),
                require('../components/login/login.lang.en.json'),
                require('../components/message/message.lang.en.json'),
                require('../components/menu/menu.lang.en.json'),
                require('../components/option/option.lang.en.json'),
                require('../components/dialog/dialog.lang.en.json'),
                require('../components/pagination/pagination.lang.en.json'),
                require('../components/phone-number/phone-number.lang.en.json'),
                require('../components/repeater/repeater.lang.en.json'),
                require('../components/rich-text-editor/rich-text-editor.lang.en.json'),
                require('../components/table/table.lang.en.json'),
                require('../components/toast/toast.lang.en.json'),
                require('../components/tree/tree.lang.en.json'),
                require('../components/scroll-top/scroll-top.lang.en.json'),
                require('../components/show-more/show-more.lang.en.json'),
                require('../components/sidebar/sidebar.lang.en.json'),
                require('../components/spinner/spinner.lang.en.json'),
                require('../components/switch/switch.lang.en.json'),
                require('../components/textfield/textfield.lang.en.json'),
                require('../components/timepicker/timepicker.lang.en.json'),
                require('../components/tooltip/tooltip.lang.en.json'),
                require('../components/validation-message/validation-message.lang.en.json'),
                require('../filters/filesize/filesize.lang.en.json'),
                require('../filters/date/date/date.lang.en.json'),
                require('../filters/date/period/period.lang.en.json'),
                require('../filters/date/time/time.lang.en.json'),
                require('../filters/date/date-time/date-time.lang.en.json')
            ];

            msgs.forEach(msgs => i18n.addMessages(ENGLISH, msgs));
        } else {
            throw new Error(
                'EnglishPlugin.install -> You must use the i18n plugin.'
            );
        }
    }
};

export default EnglishPlugin;
