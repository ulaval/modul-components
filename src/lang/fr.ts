import { PluginObject } from 'vue';

import { BundleMessagesMap, FRENCH, Messages } from '../utils/i18n/i18n';

const FrenchPlugin: PluginObject<any> = {
    install(v, options): void {
        console.debug('FrenchPlugin', 'plugin.install');
        const i18n: Messages = (v.prototype as any).$i18n;
        if (i18n) {
            const msgs: BundleMessagesMap[] = [
                require('../components/accordion/accordion.lang.fr.json'),
                require('../components/accordion-group/accordion-group.lang.fr.json'),
                require('../components/list-item/list-item.lang.fr.json'),
                require('../components/dialog/dialog.lang.fr.json'),
                require('../components/datefields/datefields.lang.fr.json'),
                require('../components/datepicker/datepicker.lang.fr.json'),
                require('../components/dropdown/dropdown.lang.fr.json'),
                require('../components/dropdown-item/dropdown-item.lang.fr.json'),
                require('../components/dropdown-group/dropdown-group.lang.fr.json'),
                require('../components/edit-window/edit-window.lang.fr.json'),
                require('../components/file-select/file-select.lang.fr.json'),
                require('../components/file-upload/file-upload.lang.fr.json'),
                require('../components/input-style/input-style.lang.fr.json'),
                require('../components/link/link.lang.fr.json'),
                require('../components/limit-text/limit-text.lang.fr.json'),
                require('../components/message/message.lang.fr.json'),
                require('../components/menu/menu.lang.fr.json'),
                require('../components/modal/modal.lang.fr.json'),
                require('../components/phone-number/phone-number.lang.fr.json'),
                require('../components/scroll-top/scroll-top.lang.fr.json'),
                require('../components/sidebar/sidebar.lang.fr.json'),
                require('../components/spinner/spinner.lang.fr.json'),
                require('../components/switch/switch.lang.fr.json'),
                require('../components/textfield/textfield.lang.fr.json'),
                require('../components/timepicker/timepicker.lang.fr.json'),
                require('../components/tooltip/tooltip.lang.fr.json'),
                require('../components/validation-message/validation-message.lang.fr.json')
            ];

            msgs.forEach(msgs => i18n.addMessages(FRENCH, msgs));
        } else {
            throw new Error(
                'FrenchPlugin.install -> You must use the i18n plugin.'
            );
        }
    }
};

export default FrenchPlugin;
