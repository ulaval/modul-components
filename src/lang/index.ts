/**
 * List public declaration here.
 */
export * from './fr';
