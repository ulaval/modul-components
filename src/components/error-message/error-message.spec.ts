import Vue from 'vue';
import { mount, Wrapper } from '@vue/test-utils';
import { addMessages } from '../../../tests/helpers/lang';
import { renderComponent, WrapChildrenStub } from '../../../tests/helpers/render';
import ErrorMessagePlugin, { MErrorMessage } from './error-message';
import uuid from '../../utils/uuid/uuid';
import moment from 'moment';

jest.mock('../../utils/uuid/uuid');
(uuid.generate as jest.Mock).mockReturnValue('uuid');

describe('MErrorMessage', () => {
    let userAgent: string = window.navigator.userAgent;
    beforeEach(() => {
        Vue.use(ErrorMessagePlugin);
        addMessages(Vue, ['components/error-message/error-message.lang.fr.json']);

        userAgent = window.navigator.userAgent;
        Object.defineProperty(window.navigator, 'userAgent', { value: 'modul-user-agent', configurable: true });
    });

    afterEach(() => {
        Object.defineProperty(window.navigator, 'userAgent', { value: userAgent, configurable: true });
    });

    it('should render correctly collapsed', () => {
        const error: Wrapper<MErrorMessage> = mount(MErrorMessage, {
            localVue: Vue
        });

        return expect(renderComponent(error.vm)).resolves.toMatchSnapshot();
    });

    it('should render correctly expanded', () => {
        const error: Wrapper<MErrorMessage> = mount(MErrorMessage, {
            localVue: Vue,
            propsData: {
                date: moment('2018-01-02T00:01:02')
            },
            stubs: {
                'm-accordion': WrapChildrenStub('div')
            }
        });

        return expect(renderComponent(error.vm)).resolves.toMatchSnapshot();
    });
});
