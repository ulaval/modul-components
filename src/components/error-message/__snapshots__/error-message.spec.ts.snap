// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MErrorMessage should render correctly collapsed 1`] = `
<div>
    <div class="m-message m--has-error m--is-skin-regular">
        <div class="m-message__wrap">
            <div class="m-message__icon">
                <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                    <use xlink:href="#error" aria-hidden="true"></use>
                </svg>
            </div>
            <div class="m-message__body">
                <div class="m-message__body__content"><span class="m-i18n">Un problème technique s'est produit et empêche la tâche en cours de s'achever normalement.</span></div>
            </div>
        </div>
    </div>
    <p><span class="m-i18n">Vous pouvez attendre quelques minutes et réessayer.</span></p>
    <p><span class="m-i18n">Si le problème persiste, vous pouvez contacter le soutien technique en leur fournissant le détail de l'erreur.</span></p>
    <article class="m-accordion m--is-light m--is-closed m--has-icon-left m--has-icon-large">
        <div tabindex="0" id="uuid" role="tab" aria-controls="uuid-controls" class="m-accordion__header m--has-padding">
            <div class="m-accordion__header__content"><span><span class="m-i18n">Détails de l'erreur</span></span> <span class="m-i18n m-accordion__hidden">m-accordion:open</span> </div> <span aria-hidden="true" class="m-accordion__header-icon m--is-left m--is-large m--has-border"></span></div>
    </article>
    <a href="/" class="m-link"> <span class="m-link__text"><span class="m-i18n">Pour obtenir de l'aide, contactez le soutien technique</span></span>
    </a>
</div>
`;

exports[`MErrorMessage should render correctly expanded 1`] = `
<div>
    <div class="m-message m--has-error m--is-skin-regular">
        <div class="m-message__wrap">
            <div class="m-message__icon">
                <svg aria-hidden="true" width="1em" height="1em" class="m-icon">
                    <use xlink:href="#error" aria-hidden="true"></use>
                </svg>
            </div>
            <div class="m-message__body">
                <div class="m-message__body__content"><span class="m-i18n">Un problème technique s'est produit et empêche la tâche en cours de s'achever normalement.</span></div>
            </div>
        </div>
    </div>
    <p><span class="m-i18n">Vous pouvez attendre quelques minutes et réessayer.</span></p>
    <p><span class="m-i18n">Si le problème persiste, vous pouvez contacter le soutien technique en leur fournissant le détail de l'erreur.</span></p>
    <div skin="light"><span><span class="m-i18n">Détails de l'erreur</span></span>
        <p><strong><span class="m-i18n">Navigateur</span>: </strong>modul-user-agent</p>
        <p><strong><span class="m-i18n">No référence de l'erreur</span>: </strong></p>
        <p><span class="m-i18n">Elle a eu lieu le <strong>2018-01-02</strong> à <strong>00:01:02</strong></span></p>
    </div>
    <a href="/" class="m-link"> <span class="m-link__text"><span class="m-i18n">Pour obtenir de l'aide, contactez le soutien technique</span></span>
    </a>
</div>
`;
